### DISCUSSION TITLE
LED-Ring coding in App layer

### DISCUSSION THREAD

**Discussion by quinnk4** [bitcraze/discussions#591] [Q&A] [answered]
I am trying to write a C app in the application layer using the LED ring deck, however I am unable to figure out how to address the LED deck from the app, and can't find anything in examples or previous discussions demonstrating how to do it. So far I have:

`paramVarId_t idLedRing = paramGetVarId(idLedRing)`
`//Check if decks are properly mounted`
`uint8_t ledInit = paramGetUint(idLedRing)`

All this does is check if the LED deck is attached, but I'd like to actually do something with it. Namely,

1. Is it possible to address and assign RGB values to the individual LEDs from the app layer? How?
2. Can the two 'headlight' LEDs on the front of the deck also be addressed from here?

Thanks in advance.

**Comment by knmcguire** [comment, marked as answer]
Ah I see that you already found the [internal parameter functionality](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/userguides/app_layer/#internal-log-and-param-system), very good! There is also a set functionality in there where you can set parameters

So next step would be to look at the ['Ring' parameter variables ](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/params/#ring)which will be intersting for you. 

1. If you want to assign RGB colors to the LED, first put[ the effect](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/params/#ringeffect) to solid color, and then set the [RGB colors](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/params/#ringsolidred) one by on .
2. For the headlights, the [ring.headlightEnable parameter](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/params/#ringheadlightenable) is the one you want.

hope this helps!

**Reply by quinnk4** [reply]
Thanks, this was helpful. The code I ended up using was:

`paramVarId_t ledeffect = paramGetVarId("ring", "effect");`
`paramVarId_t solidred= paramGetVarId("ring", "solidRed");`
`paramVarId_t solidblu= paramGetVarId("ring", "solidBlue");`
`paramVarId_t solidgrn= paramGetVarId("ring", "solidGreen");`
`uint8_t solidColourEffect = 7;`
`uint8_t high_value = 20;`
`uint8_t low_value = 0;`

`paramSetInt(ledeffect, solidColourEffect);`
`paramSetInt(solidgrn, low_value);`
`paramSetInt(solidblu, low_value);`
paramSetInt(solidred, high_value);`

Haven't finished much more than basic testing, but this is pretty much what I was looking for, thanks!
